content/smyja/how-to-use-graphql-with-django.md

```diff
@@ -0,0 +1,436 @@
+
+This tutorial will cover the creation of a CRUD (create, read, update, and delete) GraphQL API for a restaurant with Django.
+
+## Properties of GraphQL
+
+The following terms are often used when interacting with GraphQL. Knowing them can be helpful, though we won't be covering all of them in this tutorial.
+
+- **Schema**: Describes the functionality available to the client applications that connect to it.
+- **Query**: A schema type that represents the `GET` request and defines the operations that can be used for reading or fetching data.
+- **Nesting**: Queries can be nested inside of other queries.
+- **Mutation**: A schema type that defines the kind of operations that can be done to modify data.
+- **Subscription**: Notifies the client server in real time about updates to the data.
+- **Resolver**: Functions that return values for fields associated with existing schema types.
+
+## Step 1: setting up our virtual environment
+
+First, we are going to create and launch a virtual environment for our project with the `virtualenv` package (which can be [installed via `pip`](https://virtualenv.pypa.io/en/latest/installation.html#via-pip). While not necessary for starting a new Django project, working in separate environments is generally a best practice that mitigates conflicts between sites. Let's open a [terminal](https://www.codecademy.com/resources/docs/general/terminal) and create a new environment named `my_env` by running the following:
+
+```bash
+virtualenv my_env
+```
+
+Next, we will activate our new environment `my_env` with either of the following commands:
+
+```bash
+# Linux/macOS
+source my_env/bin/activate
+
+# Windows
+source/scripts/activate
+```
+
+Let's go to the next step.
+
+## Step 2: creating our Django project
+
+Next, if we haven't already, let's [install](https://docs.djangoproject.com/en/4.0/topics/install/#how-to-install-django) the `Django` package. Then, let's create a new project called `restaurant_graphql_api` and change into it:
+
+```bash
+django-admin startproject restaurant_graphql_api
+cd restaurant_graphql_api
+```
+
+Next, we're going to create a new application within our project called `my_app` by running the following:
+
+```bash
+python manage.py startapp my_app
+```
+
+Then, we'll add `my_app` to our list of `INSTALLED_APPS` in our `settings.py` file under the `restaurant-graphql_api/` directory:
+
+```py
+INSTALLED_APPS = [
+  'my_app',
+  'django.contrib.admin',
+  'django.contrib.auth',
+  # ...
+]
+```
+
+## Step 3: using GraphQL with `graphene-django`
+
+To use GraphQL with Django, we will need to install the [`graphene-django`](https://docs.graphene-python.org/projects/django/en/latest/) package.
+
+```bash
+pip install graphene-django
+```
+
+This will help add GraphQL functionality to our restaurant Django app such as resolvers and mutations. Next, let's add `'graphene_django'` to the list of `INSTALLED_APPS` in our `settings.py` file:
```

Comment:
```suggestion
This will add GraphQL functionality to our restaurant Django app such as resolvers and mutations. Next, let's add `'graphene_django'` to the list of `INSTALLED_APPS` in our `settings.py` file:
```